### PR TITLE
Parsed Err msg

### LINKED ISSUE #1071: Parse error msgs on delegation
UI Version: 0.9.3

Description:

Flash error msg should have as a description "validator for this address is currently revoked" instead.

screen shot 2018-08-02 at 1 29 40 pm

### PR DESCRIPTION
**Pull request by fedekunze**
Closes #1071 

Description:

Now errors are well parsed and shows correctly on delegation attempt.

❤️ Thank you!

### CODE REVIEW

File: app/src/renderer/components/staking/PageBond.vue [hunk 1]
```diff
@@ -237,10 +237,19 @@ export default {
           })
           this.$router.push("/staking")
         } catch (err) {
-          this.$store.commit("notifyError", {
-            title: "Error While Bonding Atoms",
-            body: err.message
-          })
+          let errData = err.message.split("\n")[5]
```
Comment (faboweb) [inline app/src/renderer/components/staking/PageBond.vue]: This will probably be used more often (like with sending coins). Can we implement this somehow in `send.js` like changing the error message there?

Comment (fedekunze) [reply app/src/renderer/components/staking/PageBond.vue]: I'm trying to figure out which types of errors are encountered while transferring tokens

**Comment by faboweb** [comment]
CI seems to be down. Have you tried retriggering it?

File: test/unit/specs/components/staking/PageBond.spec.js [hunk 2]
```diff
@@ -38,6 +38,19 @@ describe("PageBond", () => {
...
+        store.commit("addToCart", {
+          id: "pubkeyZ",
+          pub_key: {
+            type: "ed25519",
+            data: "pubkeyZ"
+          },
+          voting_power: 20000,
+          shares: 75000,
+          description: "descriptionZ",
+          country: "Chile",
```
Comment (jbibla) [inline test/unit/specs/components/staking/PageBond.spec.js]: 🇨🇱

**Comment by okwme** [comment]
i updated the snapshots because the added lines looked like what i expect a revoked validator should look like. there might have also been some errors from the e2e tests yesterday that seem to be circle's fault

**Comment by fedekunze** [comment]
@okwme Le'ts take a look into them today

File: test/unit/specs/lcdClientMock.spec.js [hunk 3]
```diff
@@ -270,7 +270,7 @@ describe("LCD Client Mock", () => {
   it("executes a delegate tx", async () => {
     let stake = await client.queryDelegation(
       lcdClientMock.addresses[0],
-      lcdClientMock.validators[2]
+      lcdClientMock.validators[1]
```
Comment (faboweb) [inline test/unit/specs/lcdClientMock.spec.js]: Why was this changed?

File: test/unit/specs/components/staking/PageBond.spec.js [hunk 4]
```diff
@@ -37,19 +36,20 @@ describe("PageBond", () => {
...
-          },
-          voting_power: 20000,
-          shares: 75000,
-          description: "descriptionZ",
-          country: "Chile",
-          moniker: "aChileanValidator",
-          revoked: true
-        })
+
+        let candidates = await node.candidates()
```
Comment (faboweb) [inline test/unit/specs/components/staking/PageBond.spec.js]: candidates are directly exported from `lcdClientMock` as `validators`, no need to query them like this. we can leave it like this, though.

File: test/unit/specs/components/staking/PageBond.spec.js [hunk 5]
```diff
@@ -440,24 +440,26 @@ describe("PageBond", () => {
...
+      account: "default",
+      password: "1234567890"
     })
+
+    wrapper.update()
+    wrapper.vm.fields.bondConfirm = true
+    wrapper.vm.fields.delegates[2].revoked = true
+    wrapper.vm.fields.delegates[2].atoms = 1
     wrapper.findAll("#btn-bond").trigger("click")
-    expect(wrapper.vm.$el.querySelector(".tm-form-msg--error")).not.toBeNull()
```
Comment (faboweb) [inline test/unit/specs/components/staking/PageBond.spec.js]: shouldn't there appear an inline error rather then a notification?

File: app/src/renderer/connectors/lcdClientMock.js [hunk 6]
```diff
@@ -324,7 +324,9 @@ module.exports = {
       candidate.delegator_shares = (
         parseInt(candidate.delegator_shares) + amount
       ).toString()
-      storeTx("cosmos-sdk/MsgDelegate", tx)
+
+      // storeTx("cosmos-sdk/MsgDelegate", tx) // do we need this anymore?
```
Comment (faboweb) [inline app/src/renderer/connectors/lcdClientMock.js]: yes, this stores the txs in the txs list to be able to show them in the transaction history

**Comment by okwme** [comment]
i thought it had been removed since it wasn't defined anywhere else but I
think it's another piece missing from your PR
